akka-remote/src/main/scala/akka/remote/artery/tcp/TcpFraming.scala

```diff
@@ -88,6 +88,8 @@ import akka.util.ByteString
 
       override def parse(reader: ByteReader): ParseResult[EnvelopeBuffer] = {
         val frameLength = reader.readIntLE()
+        if (frameLength > maxFrameLength)
+          throw new FramingException(s"Too long frame [$frameLength], max length is [$maxFrameLength]")
```

Comment:
I'll verify one more thing before we include this in a release. Perhaps it's not great to fail the entire inbound stream in this case. Not sure how that error is handled. Might be better to discard, if possible.

Reply:
I have confirmed that it's ok to throw FramingException. It will close that connection, but it's possible to continue fine after that.
```
outbound connection to [akka://RemoteConnectionSpec-remote-0@localhost:51094], message stream] Upstream failed, cause: StreamTcpException: The connection closed with error: Connection reset
```

It's possible to trigger such FramingException in other ways, such as wrong magic, non-matching frame length and such, so we can handle this in the same way.